### PR TITLE
Fix some Mpool / PoSt sched bugs

### PR DESCRIPTION
**Pull request by magik6k**

### CODE REVIEW

File: chain/messagepool_test.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package chain
+
+import "testing"
+
+func TestMessagePool(t *testing.T) {
```
Comment (whyrusleeping) [inline chain/messagepool_test.go]: so....

Comment (magik6k) [reply chain/messagepool_test.go]: (an attempt was made)

File: chain/messagepool.go [hunk 2]
```diff
@@ -134,13 +151,13 @@ func (mp *MessagePool) repubLocal() {
 			for _, msg := range msgs {
 				msgb, err := msg.Serialize()
 				if err != nil {
-					multierr.Append(errout, xerrors.Errorf("could not serialize: %w", err))
+					errout = multierr.Append(errout, xerrors.Errorf("could not serialize: %w", err))
```
Comment (whyrusleeping) [inline chain/messagepool.go]: nice

File: chain/messagepool.go [hunk 3]
```diff
@@ -328,15 +374,18 @@ func (mp *MessagePool) Remove(from address.Address, nonce uint64) {
...
-		//delete(mp.pending, from)
+		delete(mp.pending, from)
 	} else {
 		var max uint64
 		for nonce := range mset.msgs {
 			if max < nonce {
 				max = nonce
 			}
 		}
+		if nonce > max {
```
Comment (whyrusleeping) [inline chain/messagepool.go]: it bothers me that this if the same but backwards from the one right above it

File: storage/post.go [hunk 4]
```diff
@@ -284,19 +285,13 @@ func (m *Miner) computePost(ppe uint64) func(ctx context.Context, ts *types.TipS
 			ppe: ppe,
 			ts:  ts,
 		}).doPost(ctx)
+
+		m.scheduleNextPost(ppe + build.ProvingPeriodDuration)
```
Comment (whyrusleeping) [inline storage/post.go]: Ah, so we make sure to schedule the next one even if this one failed?

File: chain/messagepool.go [hunk 5]
```diff
@@ -328,15 +374,18 @@ func (mp *MessagePool) Remove(from address.Address, nonce uint64) {
...
+			max = nonce // we could have not seen the removed message before
+		}
+
```
Comment (Kubuxu) [inline chain/messagepool.go]: One note: we should remove messages with lower nonce than than `nonce` from function input.

File: storage/post.go (hunk 4 above)
Comment (magik6k) [reply storage/post.go]: Yep

**Pull request merged**